### PR TITLE
python3Packages.dataset: init at 1.5.2

### PR DESCRIPTION
**Pull request by xfnw**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Added a release notes entry if the change is major or breaking
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 125144` at 98e7abba run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built successfully:</summary>
  <ul>
    <li>python38Packages.dataset</li>
    <li>python39Packages.dataset</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/python-include-tests.md">python-include-tests</a>

Consider adding a `checkPhase` for tests, or if not feasible, `pythonImportsCheck`.

Near pkgs/development/python-modules/dataset/default.nix:25:0:
```
   |
25 |     description = "Toolkit for Python-based database access";
   | ^
```
</li>
  </ul>
</details>

File: pkgs/development/python-modules/dataset/default.nix [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "0q7gvs5dnhv2hah784gx2kqxzna20apphbc9k5j9vmxvn4k34ghc";
+  };
+
+  propagatedBuildInputs = [
+    sqlalchemy alembic banal
+  ];
+
+  doCheck = false;
```
Comment (legendofmiracles) [inline pkgs/development/python-modules/dataset/default.nix:22]: Why are you disabling tests? A comment would be useful.
And if there's an error, can you maybe send it here?

Comment (xfnw) [reply pkgs/development/python-modules/dataset/default.nix:22]: the tests seem to try to import a non-existent module named `test.test`

```
running build_ext
0:00:00 load avg: 0.19 Run tests sequentially
0:00:00 load avg: 0.19 [1/1] test
test test crashed -- Traceback (most recent call last):
  File "/nix/store/q6gfck5czr67090pwm53xrdyhpg6bx67-python3-3.8.9/lib/python3.8/test/libregrtest/runtest.py", line 270, in _runtest_inner
    refleak = _runtest_inner2(ns, test_name)
  File "/nix/store/q6gfck5czr67090pwm53xrdyhpg6bx67-python3-3.8.9/lib/python3.8/test/libregrtest/runtest.py", line 221, in _runtest_inner2
    the_module = importlib.import_module(abstest)
  File "/nix/store/q6gfck5czr67090pwm53xrdyhpg6bx67-python3-3.8.9/lib/python3.8/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1014, in _gcd_import
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
  File "<frozen importlib._bootstrap>", line 973, in _find_and_load_unlocked
ModuleNotFoundError: No module named 'test.test'

test failed

== Tests result: FAILURE ==

1 test failed:
    test

Total duration: 72 ms
Tests result: FAILURE
builder for '/nix/store/ina80yb0m2kk0cnw097m7rmcsr5f3anx-python3.8-dataset-1.5.0.drv' failed with exit code 2
```

**Comment by ethancedwards8** [comment]
Result of `nixpkgs-review pr 125144` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>python38Packages.dataset</li>
    <li>python39Packages.dataset</li>
  </ul>
</details>

File: pkgs/development/python-modules/dataset/default.nix (hunk 1 above)
Comment (SuperSandro2000) [reply pkgs/development/python-modules/dataset/default.nix:22]: Please add a comment with an explanation.

File: pkgs/development/python-modules/dataset/default.nix [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    inherit pname version;
+    sha256 = "0q7gvs5dnhv2hah784gx2kqxzna20apphbc9k5j9vmxvn4k34ghc";
+  };
+
+  propagatedBuildInputs = [
+    sqlalchemy alembic banal
+  ];
+
+  # checks attempt to import nonexistent module 'test.test' and fail
+  doCheck = false;
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/dataset/default.nix:23]: Please add a pythonImportsCheck.

File: pkgs/development/python-modules/dataset/default.nix [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+, buildPythonPackage
+, fetchPypi
+, sqlalchemy
+, alembic
+, banal
+}:
+
+buildPythonPackage rec {
+  pname = "dataset";
+  version = "1.5.0";
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/dataset/default.nix:11]: Latest version is 1.5.2

**Review by SuperSandro2000 — COMMENTED** [review]
otherwise LGTM

**Comment by legendofmiracles** [comment]
Result of `nixpkgs-review pr 125144` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>python310Packages.dataset</li>
    <li>python39Packages.dataset</li>
  </ul>
</details>

**Pull request merged**